Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -268,58 +284,72 @@ def status_hash(formula_or_cask, status_str, messages = nil, full_name: false, v
 
     # If a formula has to be skipped, it prints or returns a Hash contaning the reason
     # for doing so; returns false otherwise.
-    # @return [Hash, nil, Boolean]
+    sig do
+      params(
+        formula_or_cask: T.any(Formula, Cask::Cask),
+        json:            T::Boolean,
+        full_name:       T::Boolean,
+        quiet:           T::Boolean,
+        verbose:         T::Boolean,
+      ).returns(T.nilable(T.any(Hash, T::Boolean)))
+    end
     def skip_conditions(formula_or_cask, json: false, full_name: false, quiet: false, verbose: false)
-      formula = formula_or_cask if formula_or_cask.is_a?(Formula)
-      cask = formula_or_cask if formula_or_cask.is_a?(Cask::Cask)
+      case formula_or_cask
+      when Formula
+        formula = formula_or_cask
+        name = formula_name(formula, full_name: full_name)
+      when Cask::Cask
+        cask = formula_or_cask
+        name = cask_name(cask, full_name: full_name)
+      end
 
       if formula&.deprecated? && !formula.livecheckable?
         return status_hash(formula, "deprecated", full_name: full_name, verbose: verbose) if json
 
-        puts "#{Tty.red}#{formula_name(formula, full_name: full_name)}#{Tty.reset} : deprecated" unless quiet
+        puts "#{Tty.red}#{name}#{Tty.reset} : deprecated" unless quiet
         return
       end
 
       if cask&.discontinued? && !cask.livecheckable?
         return status_hash(cask, "discontinued", full_name: full_name, verbose: verbose) if json
 
-        puts "#{Tty.red}#{cask_name(cask, full_name: full_name)}#{Tty.reset} : discontinued" unless quiet
+        puts "#{Tty.red}#{name}#{Tty.reset} : discontinued" unless quiet
         return
       end
 
       if formula&.disabled? && !formula.livecheckable?
         return status_hash(formula, "disabled", full_name: full_name, verbose: verbose) if json
 
-        puts "#{Tty.red}#{formula_name(formula, full_name: full_name)}#{Tty.reset} : disabled" unless quiet
+        puts "#{Tty.red}#{name}#{Tty.reset} : disabled" unless quiet
         return
       end
 
       if formula&.versioned_formula? && !formula.livecheckable?
         return status_hash(formula, "versioned", full_name: full_name, verbose: verbose) if json
 
-        puts "#{Tty.red}#{formula_name(formula, full_name: full_name)}#{Tty.reset} : versioned" unless quiet
+        puts "#{Tty.red}#{name}#{Tty.reset} : versioned" unless quiet
         return
       end
 
-      if cask&.version&.latest? && !cask.livecheckable?
+      if cask.present? && cask.version.latest? && !cask.livecheckable?
         return status_hash(cask, "latest", full_name: full_name, verbose: verbose) if json
 
-        puts "#{Tty.red}#{cask_name(cask, full_name: full_name)}#{Tty.reset} : latest" unless quiet
+        puts "#{Tty.red}#{name}#{Tty.reset} : latest" unless quiet
         return
       end
 
-      if cask&.url&.unversioned? && !cask.livecheckable?
+      if cask.present? && cask.url.unversioned? && !cask.livecheckable?
```

Comment:
https://github.com/sorbet/sorbet/issues/957

Reply:
`url` can still be `nil` even if `cask` isn't.